fleetspeak/src/server/proto/fleetspeak_server/admin.proto

```diff
@@ -80,6 +81,15 @@ message BlacklistClientRequest {
   bytes client_id = 1;
 }
 
+message FetchClientResourceUsageRecordsRequest {
+  bytes client_id = 1;
+  uint64 limit = 2;
```

Comment:
Would it make more sense to pass a time range here instead of a limit? I don't think I can think of a use-case when I'd be interested in `100 resource usage records`, but use-cases when I'm interested in `resource usage records between 18.08.2020 and 28.08.2020`, or `resource usage records for the last 7 days` are actually quite likely.

This doesn't necessarily have to be done in this PR (especially if it's blocking the Grafana GRR PR), but I think adding a start_time and end_time parameters would make this method much more usable.

Reply:
That's an interesting perspective, which I think is worth a discussion.

I think that you are right, and passing a time range rather than a limit will be more beneficial. Using the limit like this PR does seems almost trivial with respect to google/grr#832, as Grafana passes a variable `maxDataPoints` along with its request, which easily translates into a limit for the datastore method `FetchResourceUsageRecords`. But in general, I agree that timestamps would be more efficient for Grafana as well, and allow for better querying the database.

There is one thing to consider though; implementing this will be a bit more involved, and will require either changing the existing datastore method `FetchResourceUsageRecords` to support this, or rather create a new datastore method. In either way we decide to implement it, it'll complicate Fleetspeak a little bit (in the current PR we add no new methods to the datastore interface, and only utilize existing ones).

Reply:
If you go this route, I would update the existing method, you can make it revert to the old behavior when the limit time is 0 or unset.

Somewhat as an aside, our internal datastore implementation drops most data (including resource usage records) once it is 2 weeks old, which limits the cost of returning everything. I don't remember if we ever implemented something similar for the mysql datastore, so we quite possibly did not.

Reply:
Following an offline discussion with @bgalehouse, I'll change the arguments of `FetchResourceUsageRecords` and the corresponding RPC messages to support a range (2 timestamps) rather than a limit.